### PR TITLE
ITSASU-531: Sole Trader - Accounting method Page

### PR DESCRIPTION
**Pull request by sb-dev**
Update view + controller with save & retrieve logic + content.

### CODE REVIEW

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala [hunk 1]
```diff
@@ -94,11 +97,26 @@ class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
 
     "return 303, SEE_OTHER)" when {
       "the user submits valid data" when {
+        "save and retrieve is enable" should {
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala:100]: can you add "in edit mode and there is an id value" before row 100

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala [hunk 2]
```diff
@@ -94,11 +97,26 @@ class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
...
+              FakeRequest().withFormUrlEncodedBody(modelToFormData(testAccountingMethodModel): _*)
+            )
+            status(result) mustBe SEE_OTHER
+            redirectLocation(result) mustBe
+              Some(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.SelfEmployedCYAController.show("testId").url)
+          }
+        }
+
         "not in edit mode" should {
           "redirect to sign up front end rerouting page" in withController { controller =>
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala:115]: can you add " save and retrieve is disabled" after row 114

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala [hunk 3]
```diff
@@ -109,9 +127,10 @@ class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
 
         "in edit mode" should {
           "redirect to sign up front end check your answer page" in withController { controller =>
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala:129]: add "save and retrieve is disabled" after row 128

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala [hunk 4]
```diff
@@ -135,14 +154,29 @@ class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
   "The back url" when {
     "not in edit mode" should {
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala:155]: can you add " save and retrieve is disabled" after row 155

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala [hunk 5]
```diff
@@ -135,14 +154,29 @@ class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
...
+
+      "return None when Save and Retrieve is enabled" in withController { controller =>
+        enable(SaveAndRetrieve)
         mockAuthSuccess()
-        controller.backUrl(false) mustBe routes.BusinessListCYAController.show().url
+        controller.backUrl(id = None, isEditMode = false) mustBe None
       }
     }
     "in edit mode" should {
       "return a url for the sign up front end final CYA page" in withController { controller =>
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala:169]: can you add " save and retrieve is disabled" after row 168

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala [hunk 6]
```diff
@@ -135,14 +154,29 @@ class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
...
+        disable(SaveAndRetrieve)
+        mockAuthSuccess()
+        controller.backUrl(id = None, isEditMode = true) mustBe Some(appConfig.subscriptionFrontendFinalCYAController)
+      }
+
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala:174]: can you add " save and retrieve is enabled" before row 175

File: it/controllers/BusinessAccountingMethodControllerISpec.scala [hunk 7]
```diff
@@ -96,11 +98,49 @@ class BusinessAccountingMethodControllerISpec extends ComponentSpecBase {
...
+          Then("Should return a SEE_OTHER with a redirect location of routing controller in Subscription FE")
+          res must have(
+            httpStatus(SEE_OTHER),
+            redirectURI(BusinessCYAUri)
+          )
+        }
+      }
+
       "not in edit mode" should {
         "redirect to sign up frontend business routing controller" in {
```
Comment (scullythornton1) [inline it/controllers/BusinessAccountingMethodControllerISpec.scala:138]: can you add " save and retrieve is disabled" after row 137

File: it/controllers/BusinessAccountingMethodControllerISpec.scala [hunk 8]
```diff
@@ -119,6 +159,8 @@ class BusinessAccountingMethodControllerISpec extends ComponentSpecBase {
           Given("I setup the Wiremock stubs")
```
Comment (scullythornton1) [inline it/controllers/BusinessAccountingMethodControllerISpec.scala:159]: can you add "save and retrieve is disabled" before row 158

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala [hunk 9]
```diff
@@ -120,13 +127,30 @@ class BusinessAccountingMethodViewSpec extends ViewSpec {
     }
 
     "have a continue button when it is not in edit mode" in {
+      disable(SaveAndRetrieve)
       document().getButtonByClass mustBe BusinessAccountingMethodMessages.continue
     }
 
     "have an update button when it is in edit mode" in {
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala:134]: can you add" save and retrieve is disabled"

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala [hunk 10]
```diff
@@ -120,13 +127,30 @@ class BusinessAccountingMethodViewSpec extends ViewSpec {
     }
 
     "have a continue button when it is not in edit mode" in {
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala:129]: can you add" save and retrieve is disabled"

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala [hunk 11]
```diff
@@ -141,18 +165,20 @@ class BusinessAccountingMethodViewSpec extends ViewSpec {
     ).mustHaveGovUkErrorNotificationMessage(emptyError)
   }
```
Comment (scullythornton1) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala:166]: can you add a test for back link for two different scenarios:
1. when showing 
2. not showing

Comment (sb-dev) [reply test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/views/BusinessAccountingMethodViewSpec.scala:166]: The tests are on line 64 and line 69.

**Pull request merged**